internal/metrics/metrics.go

```diff
@@ -0,0 +1,43 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/go-errors/errors"
+	"github.com/prometheus/common/expfmt"
+)
+
+// Metrics required to be sent to load balancer
+type Metrics struct {
+	PeerCount             *float64
+	BestBlockHeight       *float64
+	FinalizedBlockHeight  *float64
+	ReadyTransactionCount *float64
+}
+
+const (
+	metricsEndpoint = "/metrics"
+)
+
+// GetNodeMetrics retrieves polkadot metrics from prometheus server
+func GetNodeMetrics(baseURL string) (*Metrics, error) {
+	resp, err := http.Get(baseURL + metricsEndpoint)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, errors.New("Metrics endpoint returned error")
```

Comment:
I think these cases should be separated. Reason being that when this `Metrics endpoint returned error` error is raised, there is no context about error inside message. My suggestion is something like this:

```go
if err != nill {
 // return original error 
} else if resp.StatusCode != http.StatusOk {
 // return new error and add status code to error message
}
```